Sources/FutureType.swift

```diff
@@ -30,14 +30,14 @@ public protocol FutureType: CustomDebugStringConvertible, CustomReflectable {
     /// A type that represents the result of some asynchronous operation.
     associatedtype Value
 
-    /// Call some function once the value is determined.
+    /// Call some function `body` once the value is determined.
     ///
-    /// If the value is determined, the function should be submitted to the
-    /// queue immediately. An `upon` call should always execute asynchronously.
+    /// If the value is determined, the function should be submitted to
+    /// to the `executor` immediately.
```

Comment:
This should come from documenting the intent of `ExecutorType`. Unfortunately, we can only be as strong about an `upon` always being asynchronous for the queue overloads; though I don't recommend it, you can totally write an `ExecutorType` that just calls its closure immediately.